src/collections/Form/Form.js

```diff
@@ -95,9 +95,10 @@ class Form extends Component {
   static TextArea = FormTextArea
 
   handleSubmit = e => {
-    const { action, onSubmit } = this.props
-    if (!action) e.preventDefault()
-    if (onSubmit) onSubmit(e, { ...this.props })
+    const { action } = this.props
+
+    if (!action) _.invoke(e, 'preventDefault')
```

Comment:
How about a line comment linking to the issue for this?  Looking at this later, I could see it getting reverted as it appears unnecessary.